pkg/in/tcp.go

```diff
@@ -54,8 +54,11 @@ loop:
 		lg.Debug("Stopped accepting new TCP connections. Starting to close incoming connections...", zap.String("address", l.Addr().String()))
 		wg.Wait()
 		lg.Debug("Finished previously accpted TCP connections.", zap.String("address", l.Addr().String()))
+		connWG.Done()
+	} else {
+		wg.Wait()
+		connWG.Done()
 	}
-	connWG.Done()
```

Comment:
would it make sense to keep connWG.Done unconditional and introduce the else statement for wg.Wait?

`} else {
		wg.Wait()
		
	}
connWG.Done()`

Reply:
not sure what you mean, @azhiltsov 
both wg.Wait() and  connWG.Done() are unconditional.